tests/integration/devfile/cmd_devfile_catalog_test.go

```diff
@@ -79,4 +81,150 @@ var _ = Describe("odo devfile catalog command tests", func() {
 			helper.MatchAllInOutput(output, []string{"name: nodejs-starter", "Registry: " + registryName})
 		})
 	})
+	Context("When executing catalog describe component with a component name that does not have a devfile component", func() {
+		It("should print message that there is no Odo devfile component available", func() {
+			output := helper.CmdShouldPass("odo", "catalog", "describe", "component", "java")
+			helper.MatchAllInOutput(output, []string{"There are no Odo devfile components with the name \"java\""})
+		})
+	})
+	Context("When executing catalog describe component with more than one argument", func() {
+		It("should give an error saying it received too many arguments", func() {
+			output := helper.CmdShouldFail("odo", "catalog", "describe", "component", "too", "many", "args")
+			helper.MatchAllInOutput(output, []string{"accepts 1 arg(s), received 3"})
+		})
+	})
+	Context("When executing catalog describe component with no arguments", func() {
+		It("should give an error saying it expects exactly one argument", func() {
+			output := helper.CmdShouldFail("odo", "catalog", "describe", "component")
+			helper.MatchAllInOutput(output, []string{"accepts 1 arg(s), received 0"})
+		})
+	})
+
+	Context("When executing catalog list components with experimental mode set to true", func() {
+		It("should prove that nodejs is present in both S2I Component list and Devfile Component list", func() {
+			output := helper.CmdShouldPass("odo", "catalog", "list", "components", "-o", "json")
+			cmpName := []string{"nodejs"}
+			err := utils.VerifyCatalogListComponent(output, cmpName)
+			Expect(err).Should(BeNil())
+		})
+	})
+
+	Context("odo devfile catalog command tests without kubeconfig", func() {
+		JustBeforeEach(func() {
+			helper.LocalKubeconfigSet("")
+		})
+
+		It("should list all supported devfile components", func() {
+			output := helper.CmdShouldPass("odo", "catalog", "list", "components")
+			wantOutput := []string{
+				"Odo Devfile Components",
+				"NAME",
+				"java-springboot",
+				"java-openliberty",
+				"java-quarkus",
+				"DESCRIPTION",
+				"REGISTRY",
+				"DefaultDevfileRegistry",
+			}
+			helper.MatchAllInOutput(output, wantOutput)
+		})
+
+		It("should list devfile components in json format with -o flag", func() {
+			output := helper.CmdShouldPass("odo", "catalog", "list", "components", "-o", "json")
+			wantOutput := []string{
+				"odo.dev/v1alpha1",
+				"devfileItems",
+				"java-openliberty",
+				"java-springboot",
+				"nodejs",
+				"java-quarkus",
+				"java-maven",
+			}
+			helper.MatchAllInOutput(output, wantOutput)
+		})
+	})
+
+	Context("When executing catalog list components with registry that is not set up properly, without kubeconfig set", func() {
```

Comment:
Again scenarios like https://github.com/openshift/odo/pull/3935/files#diff-318517042700080474192fc7dc04e456R147-R229 should be mentioned in #4052 for unit test coverage which will be considered after we do refactoring